crates/lsp_test/src/lsp_client.rs

```diff
@@ -117,15 +122,18 @@ impl TestClient {
 
     pub async fn shutdown(&mut self) {
         // TODO: Check that no messages are incoming
+        let id = self.id();
 
         // Don't use `Request::from_request()`. It has a bug with undefined
         // params (when `R::Params = ()`) which causes tower-lsp to not
         // recognise the Shutdown request.
-        let req = Request::build("shutdown").id(self.id()).finish();
+        let req = Request::build("shutdown").id(id.clone()).finish();
 
         // Unwrap: For this test client it's fine to panic if we can't send
         self.tx.send(req).await.unwrap();
-        self.recv_response().await;
+
+        let response = self.recv_response().await;
+        assert_eq!(&id, response.id());
```

Comment:
I switched `id()` to return `jsonrpc::Id` because I found it was quite useful to add in some extra asserts to ensure that a call to `recv_response()` actually returns the response you were hoping for